pkg/obfuscate/sql.go

```diff
@@ -354,6 +352,10 @@ func attemptObfuscation(tokenizer *SQLTokenizer) (*ObfuscatedQuery, error) {
 		grouping groupingFilter
 	)
 	defer metadata.Reset()
+
+	// Keep track of whether we saw a semicolon.
+	// We want to track semicolons and only emit them when they are between statements.
+	sawSemicolon := false
```

Comment:
Can you run the benchmarks and do a before/after comparison? We should do that every time we add an allocation.